### PR TITLE
Adding the Starting scene and settings menu

### PR DESCRIPTION
**Pull request by AlstonXiao**
- Players can enter scene 1 from starting scene
- Players can change settings in the settings menu
- New global settings object available to everyone

### CODE REVIEW

**Review by zyd2001 — DISMISSED** [review]
nice

File: Assets/Script/Backend/Settings/GlobalSettings.cs [hunk 1]
```diff
@@ -1,25 +1,176 @@
...
+
+        public void SaveUserDefinedSettings()
+        {
+            if (System.IO.File.Exists(userSettingsPath))
+            {
+                File.Delete(userSettingsPath);
+            }
+            var yaml = serializer.Serialize(userDefinedSettings);
+            StreamWriter writer = new StreamWriter(userSettingsPath, false);
+            writer.Write(yaml);
```
Comment (jialuogao) [inline Assets/Script/Backend/Settings/GlobalSettings.cs]: Use WriteAllText to overwrite file
https://learn.microsoft.com/en-us/dotnet/api/system.io.file.writealltext?view=net-7.0&redirectedfrom=MSDN#System_IO_File_WriteAllText_System_String_System_String_

File: Assets/Script/Backend/Settings/GlobalSettings.cs [hunk 2]
```diff
@@ -1,25 +1,176 @@
...
+        {
+            
+        }
+
+        // Make sure it will have the user settings before calling
+        public void LoadUserDefinedSettings()
+        {
+            if (!System.IO.File.Exists(userSettingsPath))
+            {
+                throw new Exception();
```
Comment (jialuogao) [inline Assets/Script/Backend/Settings/GlobalSettings.cs]: FileNotFoundException
https://learn.microsoft.com/en-us/dotnet/api/system.io.filenotfoundexception?view=net-7.0

File: Assets/Script/Backend/Settings/GlobalSettings.cs (hunk 1 above)
Comment (AlstonXiao) [reply Assets/Script/Backend/Settings/GlobalSettings.cs]: fixed

File: Assets/Script/Backend/Settings/GlobalSettings.cs (hunk 2 above)
Comment (AlstonXiao) [reply Assets/Script/Backend/Settings/GlobalSettings.cs]: fixed

**Review by jialuogao — APPROVED** [review]

**Pull request merged**